locals.tf

```
@@ -1,3 +1,4 @@
 locals {
-  install_id = var.install_id != "" ? var.install_id : random_string.install_id.result
+  dns_project = var.dns_project != "" ? var.dns_project : var.project
```

the only suggestion I have for this is to maybe change it to `rendered_dns_project` to differentiate that it's a ternary dependent local var.

```suggestion
  rendered_dns_project = var.dns_project != "" ? var.dns_project : var.project
```